rocketmq-flink/src/main/java/org/apache/rocketmq/flink/RocketMQSource.java

```diff
@@ -291,6 +297,10 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
 
         for (Map.Entry<MessageQueue, Long> entry : offsetTable.entrySet()) {
             unionOffsetStates.add(Tuple2.of(entry.getKey(), entry.getValue()));
+            // commit offset to broker after checkpointing
+            if (enableCheckpoint) {
```

Comment:
logic is wrong here!

Reply:
you are right, the best way is do this logic when checkpoint compeleted.